pkg/clients/grpc/connect_test.go

```diff
@@ -190,3 +289,196 @@ func createTempFile(data []byte) (string, error) {
 func createTempFileHandle() (*os.File, error) {
 	return os.CreateTemp("", "test")
 }
+
+func TestVerifyAttestationReportTLS(t *testing.T) {
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err)
+
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			Organization: []string{"Test Org"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(time.Hour * 24),
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+
+	file, err := os.ReadFile("../../../attestation.bin")
+	require.NoError(t, err)
+
+	ar, err := abi.ReportCertsToProto(file)
+	require.NoError(t, err)
+	arBytes, err := proto.Marshal(ar)
+	require.NoError(t, err)
+	template.ExtraExtensions = []pkix.Extension{
+		{
+			Id:    customSEVSNPExtensionOID,
+			Value: append(arBytes, make([]byte, attestationReportSize-len(arBytes))...),
+		},
+	}
+
+	certDER, err := x509.CreateCertificate(rand.Reader, &template, &template, &privateKey.PublicKey, privateKey)
+	require.NoError(t, err)
+
+	backendinfoFile, err := os.ReadFile("../../../scripts/backend_info/backend_info.json")
+	require.NoError(t, err)
+
+	attestationConfiguration = AttestationConfiguration{
+		SNPPolicy:   &check.Policy{},
+		RootOfTrust: &check.RootOfTrust{},
+	}
+	err = json.Unmarshal(backendinfoFile, &attestationConfiguration)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name     string
+		rawCerts [][]byte
+		err      error
+	}{
+		{
+			name:     "Valid certificate with attestation, malformed guest policy",
```

Comment:
this would require fetching an attestation report during the test since the expected report data is the public key, so this would not be possible here